generated/nidcpower/nidcpower/lcr_measurement.py

```diff
@@ -0,0 +1,229 @@
+            **Defined Values**:
+
+            +-----------------------+-----------------------------------------------------+
+            | InstrumentMode.SMU_PS | The channel(s) are operating as a power supply/SMU. |
+            +-----------------------+-----------------------------------------------------+
+            | InstrumentMode.LCR    | The channel(s) are operating as an LCR meter.       |
+            +-----------------------+-----------------------------------------------------+
+
+        in_compliances (InCompliances): Indicates whether the output was in DC compliance and/or
+            AC compliance at the time the measurement was taken.
+
+        unbalanced (bool): Indicates whether the output was unbalanced at the time the
+            measurement was taken.
+    """
+    LCR = namedtuple(typename="LCR", field_names=("inductance", "capacitance", "resistance"))
+    InCompliances = namedtuple(typename="InCompliances", field_names=("dc", "ac"))
+
+    _lcr_measurement_field_metadata = [
+        # field_name           label(s)
+        ("channel"           , "Channel"                                                             ),  # noqa: E202,E203
+        ("vdc"               , "DC voltage"                                                          ),  # noqa: E202,E203
+        ("idc"               , "DC current"                                                          ),  # noqa: E202,E203
+        ("stimulus_frequency", "Stimulus frequency"                                                  ),  # noqa: E202,E203
+        ("ac_voltage"        , "AC voltage"                                                          ),  # noqa: E202,E203
+        ("ac_current"        , "AC current"                                                          ),  # noqa: E202,E203
+        ("z"                 , "Impedance"                                                           ),  # noqa: E202,E203
+        ("series_lcr"        , ("Series inductance", "Series capacitance", "Series resistance")      ),  # noqa: E202,E203
+        ("parallel_lcr"      , ("Parallel inductance", "Parallel capacitance", "Parallel resistance")),  # noqa: E202,E203
+        ("d"                 , "Dissipation factor"                                                  ),  # noqa: E202,E203
+        ("measurement_mode"  , "Measurement mode"                                                    ),  # noqa: E202,E203
+        ("in_compliances"    , ("DC in compliance", "AC in compliance")                              ),  # noqa: E202,E203
+        ("unbalanced"        , "Unbalanced"                                                          ),  # noqa: E202,E203
+    ]
+
+    def __init__(self, data):
+        """LCRMeasurement
+
+        Creates and returns an LCRMeasurement object.
+
+        Args:
+            data (struct_NILCRMeasurement): The LCR measurement ctypes object returned by the driver.
+        """
+        self.channel = ""
```

Comment:
consider making channel a parameter to __init__

Reply:
The `channel` field would not be populated through the constructor as `LCRMeasurement` objects would be converted directly from the `struct_NILCRMeasurement` objects returned by the driver before the additional Python code retrieves the channel names separately from the driver and populate the `channel` field of the already-constructed `LCRMeasurement` objects accordingly. Kindly refer to https://github.com/ni/nimi-python/pull/1771/files#diff-31582e14c636470681a52cfa45ebf6f28dbd68afde8827d6058a43022b3231c4R4921 for more details of how this is done.

Based on your previous comments, we should remove code paths that would not be used, so I believe `channel` should not be made a parameter to `__init__()`.

Reply:
This is going to get a bit philosophical. Per OOP best practices, an object should be in an internally consistent state after construction.

I argue that a channel of "" puts the object in an internally inconsistent state so it'd be nice to avoid.

If it is reasonable to do so, see if the client code can delay construction of the LCRMeasurement until it has the channel. If it is not reasonable to do so because it complicates the code, then reply and resolve this again.

Reply:
The current implementation for `fetch_multiple_lcr` and `measure_multiple_lcr` has the public method calls into the private method (which is codegen-ed) to retrieve and construct the LCRMeasurement instances before populating the `channel` fields with the respective channel names in the public method. The codegen logic does not currently support passing in additional parameter(s) in the conversion from the ctypes instances to the Python instances of `LCRMeasurement`, so it will be very complicated to delay the construction of the LCRMeasurement until it has the channel.